use up-to-date engine262.js lib

caub opened the pull request:
Either #44 or this PR

I hope one can work, to keep engine262 more up to date

The current vendor/engine262.js is more than 1 year old

caub commented:
@brigand I don't understand why github actions fail

brigand commented:
I can't run `npm install` locally either, but it's updating my package-lock.json. Weird...

caub commented:
for me `npm i` works, it modifies package-lock too

```
caub@inspiron:~/dev/jellobot (sync-engine262)$ npm i
> node-icu-charset-detector@0.2.0 install /home/caub/dev/jellobot/node_modules/node-icu-charset-detector
> node-gyp rebuild

make: Entering directory '/home/caub/dev/jellobot/node_modules/node-icu-charset-detector/build'
  CXX(target) Release/obj.target/node-icu-charset-detector/node-icu-charset-detector.o
../node-icu-charset-detector.cpp:7:10: fatal error: unicode/ucsdet.h: No such file or directory
 #include <unicode/ucsdet.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
node-icu-charset-detector.target.mk:109: recipe for target 'Release/obj.target/node-icu-charset-detector/node-icu-charset-detector.o' failed
make: *** [Release/obj.target/node-icu-charset-detector/node-icu-charset-detector.o] Error 1
make: Leaving directory '/home/caub/dev/jellobot/node_modules/node-icu-charset-detector/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 4.15.0-115-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/caub/dev/jellobot/node_modules/node-icu-charset-detector
gyp ERR! node -v v12.18.3
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: node-icu-charset-detector@0.2.0 (node_modules/node-icu-charset-detector):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: node-icu-charset-detector@0.2.0 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

added 205 packages from 125 contributors, removed 190 packages, updated 256 packages and audited 875 packages in 15.362s

10 packages are looking for funding
  run `npm fund` for details

found 76673 vulnerabilities (76638 low, 29 moderate, 6 high)
  run `npm audit fix` to fix them, or `npm audit` for details

caub@inspiron:~/dev/jellobot (sync-engine262 *)$ echo $?
0

```

brigand commented:
Fixed the initial issue. Now it's failing on the engine262 test.

```

    Expected: "(okay) '1'"
    Received: "(fail) Error: EACCES: permission denied, open '/run/node_modules/engine262.js'"

      122 |   it('works with engine262', async () => {
      123 |     const output = await testEval('e> ({foo: 1})?.foo ?? 2');
    > 124 |     expect(output).toEqual(`(okay) '1'`);
          |                    ^
      125 |   });
      126 | });
      127 | 

      at Object.toEqual (src/plugins/js-eval/__tests__/jsEvalPlugin-test.js:124:20)
```

caub commented:
thanks, will check it, maybe we should cherry-pick your fixes in #44 which is maybe better

brigand commented on src/plugins/js-eval/run.js:
```diff
@@ -78,18 +76,25 @@ const run = async (code, environment, timeout) => {
-      runWithEngine262 = (code) => inspect(realm.evaluateScript(code), realm);
-    }
-    return runWithEngine262(code);
+    return new Promise((resolve, reject) => {
+      realm.scope(() => {
+        // const print = new Value((args) => {
+        //   console.log(...args.map((tmp) => _inspect(tmp)));
+        //   return Value.undefined;
+        // });
+        // CreateDataProperty(realm.GlobalObject, new Value('print'), print);
+        resolve(_inspect(realm.evaluateScript(code)));
+      });
+    });
```
I'm curious about the failure cases here. Could you add tests for a syntax error, and runtime error with engine262?

caub replied:
it's not ideal https://github.com/brigand/jellobot/pull/45/commits/e3ae50829c6df9d96ae654af2005324d23e414d9 because I don't know how t differentiate errors from `realm.evaluateScript(code)`

caub replied:
Ok, it looks better now: https://github.com/brigand/jellobot/pull/45/commits/dca7f53452eb3535e02476e3264ce8b204a12d63
